femtools/Detector_Tools.F90

```diff
@@ -253,6 +309,27 @@ subroutine insert_into_detector_list(detector, current_list)
 
   end subroutine insert_into_detector_list
 
+  subroutine insert_into_temp_detector_list(detector, current_list)
```

Comment:
The purpose of the temp_lists is to create a second 'temporary' particle lists only containing the particles within a certain particle group that exist on a specific node of the mesh (i.e. the 'node_particles' lists). Constructing these temporary lists at the beginning allows us to loop over only these particles when performing spawning/deleting instead of going through the entire particle list and checking if each particle falls within the control volume for each routine. 

The temp features are also necessary to keep the original particle lists intact when creating these new node lists and looping over them. For example, if we pick the first particle on a node and want to loop from there, we would originally have needed to use particle%next. The problem is, this will take us to the next particle on the particle_group list, not the node list. As a solution, I've added the %temp_next parameter to link particles in the node_list without interfering with how they are linked on the particle_group list.

Reply:
Great, that helps a lot! I guess I was missing the node specificity requirement.